### PR TITLE
samples: add reservation samples

### PR DESCRIPTION
**Pull request by anguillanneuf**
Towards b/213558019
Adding Python samples for the following region tags:

- `pubsublite_create_reservation`
- `pubsublite_get_reservation`
- `pubsublite_update_reservation`
- `pubsublite_list_reservations`
- `pubsublite_delete_reservation`

Java already added in https://github.com/googleapis/java-pubsublite/pull/1028

### CODE REVIEW

File: samples/snippets/create_lite_reservation_example.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# Copyright 2022 Google Inc. All Rights Reserved.
```
Comment (hongalex) [inline samples/snippets/create_lite_reservation_example.py:3]: This header should just be:

`Copyright 2022 Google LLC`. See go/releasing/preparing#Apache-header

Please update here and in the other new files created in this PR.

File: samples/snippets/create_lite_reservation_example.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+def create_lite_reservation(
+    project_number, cloud_region, reservation_id, throughput_capacity,
+):
+    # [START pubsublite_create_reservation]
+    from google.api_core.exceptions import AlreadyExists
+    from google.cloud.pubsublite import AdminClient, Reservation
+    from google.cloud.pubsublite.types import CloudRegion, ReservationPath
+
+    # TODO(developer):
+    # project_number = 1122334455
```
Comment (hongalex) [inline samples/snippets/create_lite_reservation_example.py:34]: Does python support using project IDs? If so, should we prefer this over project numbers?

**Review by hongalex — APPROVED** [review]
for the most part LGTM, some issues with license headers and question about project numbers vs IDs

File: samples/snippets/create_lite_reservation_example.py (hunk 2 above)
Comment (anguillanneuf) [reply samples/snippets/create_lite_reservation_example.py:34]: To be consistent with the UI, Pub/Sub Lite uses project numbers. It's true of all existing Java/Go samples. 
image

File: samples/snippets/create_lite_reservation_example.py (hunk 1 above)
Comment (anguillanneuf) [reply samples/snippets/create_lite_reservation_example.py]: I updated the license headers for new files created by this PR. I can change the rest in a separate PR. Thanks for spotting this one.

**Review by dandhlee — APPROVED** [review]
LGTM!

**Pull request merged**